proxy/src/main/java/com/velocitypowered/proxy/connection/client/HandshakeSessionHandler.java

```diff
@@ -152,6 +153,13 @@ private void handleLogin(Handshake handshake, InitialInboundConnection ic) {
   }
 
   private ConnectionType getHandshakeConnectionType(Handshake handshake) {
+    if (handshake.getServerAddress().contains(ModernForgeConstants.MODERN_FORGE_TOKEN)
+            && handshake.getProtocolVersion().compareTo(ProtocolVersion.MINECRAFT_1_20_2) >= 0) {
+      String host = handshake.getServerAddress();
+      String[] split = host.split("\0", 2);
+      ModernForgeConstants.MODERN_FORGE_HOSTNAME_TOKEN = "\0" + split[1];
```

Comment:
Now you're leaking connection state globally... should probably also mirror forges logic of reading the number out and storing that, then it can be reconstructed later

Reply:
Corrected, hoping this is the expected way

Reply:
You are still leaking the state. Just initialize a connection type instance every time. https://github.com/ArclightPowered/lightcity/blob/master/src/main/java/io/izzel/arclight/lightcity/ModernForgeConnectionType.java

Reply:
forge does not process GameProfile, nor does it forward GameProfile in modern mode.

Reply:
@IzzelAliz Hey , is it possible approving the PR without this thing or is it necessary for the future?

Reply:
it's neccesary, otherwise it's just half baked support that will break as soon as forge touches anything with it

Reply:
> it's neccesary, otherwise it's just half baked support that will break as soon as forge touches anything with it

Oh ok I understood

Reply:
I've tried my best to fix it